orttraining/orttraining/training_api/checkpoint.cc

```diff
@@ -1,581 +1,720 @@
 // Copyright (c) Microsoft Corporation. All rights reserved.
 // Licensed under the MIT License.
 
```

Comment:
A general question is why we plan to remove the protobuf ckpt instead of having it as another format?

Reply:
There are primarily two reasons:

1. Reading and writing of protobuf messages is disallowed in a minimal build. We needed to come up with a way to support the On-Device Training minimal build. One option was to use flatbuffer format for the minimal build and the tensor proto format for the non-minimal build. But this could potentially create confusion and probably not the best solution from a design perspective. Since flatbuffer is supported in both the full build as well as the minimal build, it made sense to just go ahead with that.
2. We wanted to move away from having a directory structure with multiple files inside and rely on extracting the checkpoint based on the naming of the file. I think we had discussed this earlier in a call earlier as well.